@here/harp-materials/lib/ShaderChunks/ExtrusionChunks.ts

```diff
@@ -6,14 +6,46 @@
 
 export default {
     extrusion_pars_vertex: `
-attribute vec3 extrusionAxis;
+attribute vec4 extrusionAxis;
 uniform float extrusionRatio;
+varying vec4 vExtrusionAxis;
 `,
     extrusion_vertex: `
-transformed = transformed - extrusionAxis + extrusionAxis * extrusionRatio;
+transformed = transformed - extrusionAxis.xyz + extrusionAxis.xyz * extrusionRatio;
+vExtrusionAxis = vec4(normalMatrix * extrusionAxis.xyz, extrusionAxis.w);
+`,
+    extrusion_normal_fragment_begin: `
+#ifdef FLAT_SHADED
+    vec3 normal = vec3(0.0);
+    if (vExtrusionAxis.w > 0.999999) {
+        normal = normalize(vExtrusionAxis.xyz);
```

Comment:
Please add a comment what you are doing here!

Reply:
Done.